### ISSUE TITLE
lint: typeUnparen false positive on interface assertion idiom

### ISSUE THREAD

**Issue by Quasilyte** [go-critic/go-critic#258]
Reproducer:
```
gocritic check-package -enable=typeUnparen io
```
Output:
```
io/multi.go:72:22: typeUnparen: could simplify (*multiWriter)(nil) to (*multiWriter)(nil)
```
Code:
```go
var _ stringWriter = (*multiWriter)(nil)
```
In other words, it fails on interface assertion idiom.

negative test suite should be extended as well.

**Comment by Quasilyte** [comment]
Can't reproduce. Seems to be fixed now.
There are other issues with typeUnparen though. Will send PR  soon.
